apps/dashboard/src/data/projects/controls.js

```diff
@@ -1,11 +1,46 @@
 /**
  * Internal dependencies
  */
-import { createProject, updateProject } from '@crowdsignal/rest-api';
-import { PROJECT_SAVE } from '../action-types';
+// import { createProject, updateProject } from '@crowdsignal/rest-api';
+import { PROJECT_SAVE_REQUEST } from '../action-types';
```

Comment:
The comment got hidden after the rebase so re-adding it here so you don't miss it:

#33 added the `dispatchAsync` action/control which should make this obsolete.

As in, instead of dispatching `PROJECT_SAVE_REQUEST`, in your action generator, you can just:

```javascript
// Notice these are the request functions themselves, not actions
import { createProject, updateProject } from '@crowdsignal/rest-api';

...

export function* saveAndUpdateProject( projectId, project ) {

    // Dispatch PROJECT_SAVE, now used only for toggling the `saving` state.
    yield saveProject( projectId );

    try {
        // Use dispatchAsync to trigger the actual request now
        const response = yield projectId
            ? dispatchAsync( updateProject, [ projectId, project ] )
            : dispatchAsync( createProject, [ project ] ) ;

        // Any specific post processing of the response can happen here if necessary

        // Finally, update the project.
        // If we work under the assumption it's impossible for the project to update while it's saving
        // which sounds reasonable, then PROJECT_UPDATE can reset the saving flag as well which makes PROJECT_SAVE_SUCCESS obsolete.
        yield projectUpdate( response.data.id, response.data );
    } catch ( error ) {
        // Request failed
        yield saveProjectError( projectId );
    }
}
```

In fact, `dispatchAsync` should literally remove the need for any other control as it already allows for calling any asynchronous function from an action generator or resolver - the main purpose of controls - with minimal pollution to the store namespace.